geopandas/array.py

```diff
@@ -879,7 +879,10 @@ def estimate_utm_crs(self, datum_name="WGS 84"):
     def x(self):
         """Return the x location of point geometries in a GeoSeries"""
         if (self.geom_type[~self.isna()] == "Point").all():
-            return vectorized.get_x(self.data)
+            nonempty = ~self.is_empty
```

Comment:
I would maybe put this behind a if check, like

```
empty = self.is_empty
if empty.any():
    ...
else:
    return vectorized.get_x(self.data)
```

So we still have the faster path in the case there are not empty geometries

Reply:
A good suggestion, added.